### ISSUE TITLE
Find a way to read and write manifold indicators

### ISSUE THREAD

**Issue by luca-heltai** [dealii/dealii#410]
We support some mesh formats which have one *single* indicator (ucd, gmsh, may be vtk?), and so far we have always used this indicator to store and retrieve `boundary_ids`. 

Unfortunately, none of these allow the specification of **two** such indicators, to store both boundary *and* manifold id.

We could make an optional switch that by default copies the boundary indicator to the manifold indicator upon creation of a new mesh until we find a way to read and write both indicators.

Any ideas?

**Comment by tjhei** [comment]
I think it would be more flexible and cleaner to provide this as a separate function, something like ```GridTools::copy_boundary_to_manifold_id```.

**Comment by bangerth** [comment]
Or `move_boundary_to_manifold_id`, which clears the boundary id. It may require less code duplication to do it Timo's way, but I think if you introduce the function he suggests, writing the call to it from the various grid readers is not overly onerous either.

**Comment by tjhei** [comment]
> if we write a triangulation and then read it back in, they will be different, and we have no format that supports both ids...

Maybe we can look into extending the vtk reader/writer to also output the manifold id (if this is possible with the vtk legacy format). Alternatively, I am pretty sure you can output additional information with the xml based vtk format.

**Comment by bangerth** [comment]
I'm pretty sure we can write this information in gmsh format as well (I think I remember it supports attaching zero or more scalars to each mesh object). In the end, I do not think that we currently have reliable ways to save meshes and then re-read them to get the same mesh. This is particular true because no format I know of supports refined meshes. The only thing I can think of is reading the coarse grid from somewhere and then using `PersistantTriangulation`.

When I suggested `move_boundary_to_manifold_id` I meant something of the kind
```
  for (cell=...)
    for (face=...)
      if (cell->face(face)->at_boundary()) {
        cell->face(face)->set_manifold_id(cell->face(face)->boundary_id());
        cell->face(face)->set_boundary_id(numbers::invalid_boundary_id)
     }
```
This is as if we had set the manifold id right away in the mesh readers, instead of setting the boundary id.
